### PR TITLE
FIX CI - Issue #1253

### PR DESCRIPTION
**Pull request by sofiabobadilla**
Change  GitHub.connectAnonymously() in PatchFilterTest.java  to connect with Github using Jenkins environment variables instead.

### CODE REVIEW

**Review by algomaster99 — CHANGES_REQUESTED** [review]
I don't have suggestions specific to the project, but I thought I would write two cents about commit messages.

Make sure that a commit message explains what exactly it does. The first two commit messages are `'PatchFilterTest connects to Github with Jenkins field'`. There are two problems with this:
  1. I, personally, do not immediately understand what it is supposed to do.
  	1. Proposed solution: A better commit message, in my opinion, would be 'use Jenkins credentials to connect to GitHub'
  3. The commit message is same which means they both do the same thing.
  	1. It is usually nice to split commits, but not when they have the same purpose. One commit should do _one task_. It is arbitrary, however, for this case, it seems that you are doing same things for multiple invocations consolidating them into one single task. Such independent commits make reversal easy.

File: src/repairnator-realtime/src/test/java/fr/inria/spirals/repairnator/realtime/utils/PatchFilterTest.java [hunk 1]
```diff
@@ -18,8 +18,9 @@ public class PatchFilterTest {
     public void testSingleLineCommit() throws IOException {
         
         PatchFilter filter = new PatchFilter();
-        
-        GitHub github = GitHub.connectAnonymously();
+
+        //TODO: should use the credentials stored in Jenkins.
```
[the remark this answers is not in the document]
Comment (Sofi1410) [reply src/repairnator-realtime/src/test/java/fr/inria/spirals/repairnator/realtime/utils/PatchFilterTest.java:22]: Thanks for the suggestions, Aman 👍 .
I will keep it in mind for the next steps.

**Comment by monperrus** [comment]
Great, CI seems to fail because of https://github.com/eclipse/repairnator/issues/1254#issuecomment-1231280182

**Comment by monperrus** [comment]
CI still failing. @javierron does Jenkins already know about credential GITHUB_OAUTH?

**Comment by javierron** [comment]
@monperrus @Sofi1410 

Credentials from the test class seem to be fine.

The problem is that that the calls from [inside the tested class](https://github.com/eclipse/repairnator/blob/7c6a52f5ddf56072a3202416e33aca1c61003c12/src/repairnator-realtime/src/main/java/fr/inria/spirals/repairnator/realtime/utils/PatchFilter.java#L251) don't use the GitHub client library and are _probably_ being rate-limited (unclear from logs). We did it that way to be able "to change target URLs to a mirror in case rate limits are exceeded". 

In case rate-limits are actually the issue, we would need to either use mirrors, or the GitHub client w/ token.

**Comment by monperrus** [comment]
Thanks for the analysis.

So we have to make readFromURL configurable (with or without token), and use the configurable 
version from PatchFilterTest.

WDYT?

**Comment by Sofi1410** [comment]
The CI keeps falling with this error:
`Caused by: java.lang.ClassNotFoundException: org.codehaus.plexus.util.xml.Xpp3DomBuilder$InputLocationBuilder`

Apparently, this is a common bug related to dependencies. I will try this solution :https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=948286

**Comment by monperrus** [comment]
Yes, that's known as the "dependency hell", a topic we will research about in project Chains.

**Comment by Sofi1410** [comment]
FYI: Some tests were failing because they were using SpoonLab/sorald repository to check the runs, but the bot that was supposed to fail (codecove) is presenting some conflict therefore I change the repo to castor-software/depclean
it fulfills the requirements to be a Maven project, public and with a CI with both successful and failed runs

**Comment by monperrus** [comment]
ack. cool to see progress here.

**Comment by Sofi1410** [comment]
@andre15silva  here is the link of the test related to the dependency error we discuss at lunch: https://ci.eclipse.org/repairnator/job/repairnator-ci/job/PR-1256/44/

**Comment by monperrus** [comment]
also FTR, we discussed with @martinwitt to have a github actions, only for compiling for now (`mvn test -DskipTests`, also compiles the tests)

**Comment by Sofi1410** [comment]
Great, I'll consider this for the current failing problems.

**Comment by MartinWitt** [comment]
Hey, I'm kind of out of the loop here, do you need any help?

**Comment by MartinWitt** [comment]
Sry I haven't read your comment till now. I'm in the office normally. Feel free to come over.

**Comment by MartinWitt** [comment]
Hey Sofia, junit-4 `@Ignore` has a field Reason with String type. You could add your fixme comments inside the field. See https://junit.org/junit4/javadoc/4.13.2/org/junit/Ignore.html#value()

**Comment by sofiabobadilla** [comment]
Thanks for the suggestion :)

**Comment by sofiabobadilla** [comment]
@algomaster99 I'm getting the sorald plugin error on CI again. 
Could you share the experiments you did last time when you checked the bug?

So far the error keps being: 
- Could not download Sonar Java plugin
      -Caused by: java.nio.file.NoSuchFileException: /home/jenkins/.cache/sorald/sonar-java-plugin.jar
      
I double check the URL in sorald ([https://github.com/ASSERT-KTH/sorald/blob/380e8158e56dc5464f9e750231b47bbc43b9d94a/sorald/src/main/resources/config.properties#L1](https://repo1.maven.org/maven2/org/sonarsource/java/sonar-java-plugin/7.16.0.30901/sonar-java-plugin-7.16.0.30901.jar)) and it is working.

**Comment by sofiabobadilla** [comment]
At the same time I'm not entirely sure if it is related to the error I comment above,  but, on local, the test fr.inria.spirals.repairnator.process.inspectors.TestProjectInspector4Maven.testPatchFailingProject reproduces the same error as CI only if I don't provide the M2_HOME var, but this parameter is present o the Jenkins File so I still don't understand completely what is  going on.

**Comment by sofiabobadilla** [comment]
> > Could you share the experiments you did last time when you checked the bug?
> 
> I did my experiment [here](https://github.com/eclipse/repairnator/pull/1263). Looking back at it, I did not do anything because the problem did not persist.

I'm looking at the execution: https://ci.eclipse.org/repairnator/job/repairnator-ci/job/PR-1263/2/testReport/
And there is no info related to sorald, the test report doesn't shows result from [TestPipelineSoraldRepairTool](https://ci.eclipse.org/repairnator/job/repairnator-ci/job/PR-1256/53/testReport/junit/fr.inria.spirals.repairnator.pipeline/TestPipelineSoraldBot/TestPipelineSoraldRepairTool/). From my POV I don't see how the problem is solved. 


> 
> Successfully reaching this URL is flaky. Sometimes, it is fetched successfully, sometimes the CI just fails. I would recommend you to run you CI job again and it should pass.

I don't understand how reaching the URL could be flaky for this, could you extend a bit on this?

**Comment by algomaster99** [comment]
Now, when I look back at the report, it seems that the test pipeline timed out and it never really executed the Sorald tests. I was trying to minimise the tests run in the build and accidentally prevented execution of the Sorald tests. I cannot test it right now because I am away from keyboard.

However, the errors which you are posting it is because of failure to fetch the jar at runtime. 

> I don't understand how reaching the URL could be flaky for this, could you extend a bit on this?

It is not flaky for this. It is flaky in general. Sometimes the CI is just not able to reach the URL due to network issues. It happens in Sorald's `master` branch as well, but it is quite rare (1 in 20 builds).
The best thing to do here is to manually run it again. I also do it and 99% of the times the problem disappears if it could not fetch the jar in the previous build. Or you could have a check in the CI to run Sorald's test at least 2-3 times so that at least once you fetch the Jar.

**Comment by sofiabobadilla** [comment]
Related to last commit: https://bugs.eclipse.org/bugs/show_bug.cgi?id=575400

**Pull request merged**